src/Storages/VirtualColumnUtils.cpp

```diff
@@ -115,7 +116,19 @@ NameSet getVirtualNamesForFileLikeStorage()
     return {"_path", "_file", "_size", "_time"};
 }
 
-VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription & storage_columns)
+std::map<std::string, std::string> parseFromPath(const std::string& path)
```

Comment:
Let's mabe rename it to `parseHivePartitioningKeysAndValues`, it will be more readable